performancecopilot/pcp
# qa and find-filter

Changes committed to git@github.com:kmcdonell/pcp.git 20200403

Ken McDonell (11):
      src/find-filter: couple of small fixes
      qa/915: add _wait_for_pmproxy
      qa/common.check: fixes for _wait_for_pmproxy
      qa/915: don't restart pmproxy if it was not running at the beginning
      qa/300: sleep a bit longer
      qa/1190: (new) paranoid checking of PCP configuration
      qa/check.callback.sample: add call to ./1190 --check
      qa/common: use a more verboase annotation for the config files we change
      qa/025, qa/src/pmcdgone.c: drag these ones into the 21st century
      qa/950: after starting pmproxy, wait for it to be ready
      qa/check-flakey: (new) run some basic triage on failing tests

Mark Goodwin (1):
      qa: remake 881 for SIGUSR2

 qa/025                        |   15 ++
 qa/1190                       |  254 ++++++++++++++++++++++++++++++++++++++++++
 qa/1190.out                   |   46 +++++++
 qa/300                        |    8 -
 qa/881.out                    |    2 
 qa/915                        |   11 +
 qa/950                        |    1 
 qa/check-flakey               |  193 +++++++++++++++++++++++++++++++
 qa/check.callback.sample      |   11 +
 qa/common                     |    4 
 qa/common.check               |   52 +++++---
 qa/group                      |    1 
 qa/src/pmcdgone.c             |   59 +++++++--
 src/find-filter/.gitignore    |    1 
 src/find-filter/find-filter.c |    6 
 15 files changed, 621 insertions(+), 43 deletions(-)

Details ...

commit 0b3035e01cadc9cbf3ff188f864cb9b1b890ec3b
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Mon Apr 6 16:37:28 2020 +1000

    qa/check-flakey: (new) run some basic triage on failing tests
    
    Trying to see if they are flakey, or really a hard failure.
    
    I think I'll be using this in preference to recheck.

commit 820735775636585ca4e841aa5cad4898c090bd6d
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Mon Apr 6 15:49:43 2020 +1000

    qa/950: after starting pmproxy, wait for it to be ready

commit ac8f698fcfafa2dec44ee7c7f6736aa053324d5e
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Mon Apr 6 15:45:57 2020 +1000

    qa/025, qa/src/pmcdgone.c: drag these ones into the 21st century
    
    1. add -s flag for pmcdgone to optionally use systemctl instead of
       running the "rc" scripts directly ... more closely mimics what
       might happen in real life
    
    2. 025 sniffs around and uses -s to pmcdgone if that seems
       appropriate.

commit 5023789fd29f4435fa716e26faefa062b43bd1cc
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Mon Apr 6 15:43:19 2020 +1000

    qa/common: use a more verboase annotation for the config files we change
    
    Helps with qa/1190 to identify changes to config files that are
    expected and not a sign of badness.

commit 81545046b654c881194402169348bd594bed2246
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Mon Apr 6 15:41:56 2020 +1000

    qa/check.callback.sample: add call to ./1190 --check

commit 09647766b6b3bc7ea5d03cf3ab64d124e5b4e334
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Mon Apr 6 13:00:29 2020 +1000

    qa/1190: (new) paranoid checking of PCP configuration
    
    A desperate attempt to check for all the ways in which errant PCP QA
    test or tests can trash the PCP installation and/or confguration.
    
    Will be used in check.callback.sample to try an minimize cascading
    failures.

commit edd7b56c527d4c3d24d1b20715b377181e1690eb
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Mon Apr 6 10:29:35 2020 +1000

    qa/300: sleep a bit longer
    
    0.5 sec was not enough, and the failures were non-deterministic
    (another sign of a race in the script).
    Waiting 1.5 secs seems to have improved the situation.

commit 619ae4162a5a33e3044f5682c48b322e6d8ca1c7
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Apr 4 11:11:07 2020 +1100

    qa/915: don't restart pmproxy if it was not running at the beginning

commit 659149b9b8ba86f177807dd937a4c28f833d55fb
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Apr 4 11:09:48 2020 +1100

    qa/common.check: fixes for _wait_for_pmproxy
    
    1. timing loop was all wrong, only waiting 2 secs, not 20 secs
    2. add proper arg handling, with sane default for logfile

commit 91cd81c6500334ae52c7bd2082b69cb33420ae3e
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Apr 4 08:10:36 2020 +1100

    qa/915: add _wait_for_pmproxy

commit a257848c3fbce38edab1ba2f5b471c4bafd90b55
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri Apr 3 19:59:08 2020 +1100

    src/find-filter: couple of small fixes
    
    1. add .gitignore for executable
    2. fix cut-n-paste errors in -c time handling, exposed on older
       NetBSD (vm09)

commit f7ac80da618da80fda80ac3a6830969af6f571cd
Author: Mark Goodwin <mgoodwin@redhat.com>
Date:   Fri Apr 3 14:51:55 2020 +1100

    qa: remake 881 for SIGUSR2
